capture/split.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
```

Comment:
overall your logic to detect the file types is in the same function as you logic to parse the files. Consider spltiing into two functions

Reply:
to add a bit of explanation on this: as of now you have a lot of duplicated code that writes fastq to temporary files. Having the parsing logic in separate function that would return/yield records, or simply return an opened SeqIO handle allow you to remove a lot of redundancy to your code.